nuxeo-platform-3d-core/src/main/java/org/nuxeo/ecm/platform/threed/adapter/ThumbnailThreeDFactory.java

```diff
@@ -47,8 +46,11 @@ public Blob getThumbnail(DocumentModel documentModel, CoreSession coreSession) {
         if (!documentModel.hasFacet(THREED_FACET)) {
             throw new NuxeoException("Document is not 3D");
         }
-        Blob thumbnailBlob = (Blob) documentModel.getPropertyValue(RENDER_VIEWS_PROPERTY + "/0/thumbnail");
-
+        ThreeDDocument threeDDoc = documentModel.getAdapter(ThreeDDocument.class);
+        Blob thumbnailBlob = null;
+        if (!threeDDoc.getRenderViews().isEmpty()) {
+            thumbnailBlob = Iterables.get(threeDDoc.getRenderViews(), 0).getContent();
```

Comment:
No need for `com.google.common.collect.Iterables`, you can just call `threeDDoc.getRenderViews().iterator().next()`

Reply:
Thanks Florent. Much better.